### ISSUE TITLE
problem with a2 Settings file

### ISSUE THREAD

**Issue by lipkau** [ewerybody/a2#125]
### Expected Behavior
The user has a central file where he can/must provide configurations. Such as:
* path to `python.exe` (default is in _%path%_)
* path to `modules` (default is _modules_ in a2's root)
* path to `settings` (default is _settings_ in a2's root)
* path to `AutoHotkey.exe` (dafault is _empty_ (and a2.exe will find it in the project folder))

### Current Behavior
User has one .ahk file with all of these settings.
However, a2.ahk does not allow for the `settings\a2_settings.ahk` to be anywhere else (as it is hard-coded in _a2.ahk_ `#include settings\a2_settings.ahk`).
And if the user changes the path to `settings` (and moves the _./settings_ folder), a2 would create a new file in the default path.
_Effects on the python code is **unknown** to me for now_

### Possible Solution
1. make a _.ini_ file for convenience (for the paths):
```
[Environment]
modules = modules
AutoHotkey = 
Python = python.exe
```
2. define the version number (major, minor, revision, alpha|beta|dev) in a centralized file, so that python and ahk can read and use it. Instead of `a2_title = a2 - dev_version`
3. make debug settings part of the **a2settings_view**:
```
a2_ui_call = c:\python34\python.exe a2app.py
a2_tray_click_button = 1
a2_startup_tool_tips := true
```
4. change behavior of `a2_settings = settings`
  * [ ] fix the _settings_ path (removing the option for the user to change it)
  * [ ] allow the _settings_ path to be one of
    + ./settings  (specific for this instance)
    + %AppData%/a2/settings (specific for the current user | shared by all instances)
    + %AppDataCommon%/a2/settings (specific for this computer | shared by all users)
  * [ ] _I am open to suggestions_

### Steps to Reproduce (for bugs)
1. open `./settings/a2_settings`
2. change the value of `a2_settings = settings` to `a2_settings = c:\temp\settings`
3. quit any running a2 instances
4. move the `./settings` folder to `c:\temp\`
5. start a2.exe

**Comment by lipkau** [comment]
@ewerybody, what is your take on this?

**Comment by ewerybody** [comment]
Of course it should totally be possible to have the folders whereever you wish you want to have them! Within reason as well of course... We need to resolve any issues there.

I don't **want** to hardcode anything. You see I put a lot of variable paths already. Might very well be its not yet fool proof of course. But we're on it :]

I wonder if we should reserve a dedicated space within the users appdata area for the a2 settings.

**Comment by ewerybody** [comment]
This will be solved along with #152
I would want all of this as an optional override.
ALL of the paths should have good defaults. so

* path to `python.exe`: will be fetched from reg: (btw: not needed on release)
> `HKEY_CURRENT_USER\Software\Python\PythonCore\3.6\InstallPath>ExecutablePath`

Problem with %PATH% is: there could be ANY version given. We could of course loop the files available in %PATH% to find the right version as a fallback.

* path to `AutoHotkey.exe`: yep. This will be from the built-in lib. We "could" of course use an already installed Autohotkey.exe but there could be conflicts if the user wants to upgrade (it would be blocked by us) or uninstalls it (would break us). So as its not big anyway we should always ship an up-to-date Autohotkey executable.

* new `data` path: for settings **and** modules: By default this wil be as discussed: `appdata/local/a2` and basically not just next to the a2 dir. But the default a2 dir will also be `appdata/local/a2` so that it will by default end up just the same only that we're flexible with both locations 😃

**Comment by ewerybody** [comment]
38abb98f2deba96da4d5d0fff78d65862e72a5ab just in!
I think we got it all soved:

- [x] path to **python.exe**: will be fetched from installations found in reg
- [x] path to **modules**: in data path
- [x] path to **settings**: in data path
  - [x] data path is `appdata\local\a2\data` by default
  - [x] can be overridden in `lib\a2_config.ahk` by defining `a2data`
- [x] path to **AutoHotkey.exe**: always in `lib\AutoHotkey`
